### ISSUE TITLE
Mismatch between IDE IAR and cbuild IAR build output

### ISSUE THREAD

**Issue by tcsunhao** [Open-CMSIS-Pack/devtools#870]
**Describe the bug**
In IDE IAR, a c file is compiled into an object without '.c' sufferfix, like fsl_clock.c into fsl_clock.o
In cbuild IAR, a c file is compiled into an object with '.c' suffix, like fsl_clock.c into fsl_clock.c.o which is set in compile_commands.json.
This mismatch will result in same project build failed at least in one side, for example, If in the linker, we have 
```
initialize by copy { readwrite, section .textrw, section CodeQuickAccess, section DataQuickAccess,
  object fsl_adapter_flexspi_nor_flash.o,
  object mflash_drv.o,
  object fsl_flexspi.o,
 };
```
we want locate some object into target region, then this project cannot work in both side.

**To Reproduce**
Build a project in IDE IAR and with cbuild IAR

**Expected behavior**
IDE IAR and cbuild IAR has same output file for some input file.

**Environment (please complete the following information):**

- [tool or library] csolution/cbuild
- Version [e.g. v0.0.3] 1.6.1
- OS: [e.g. windows/linux/darwin]: Windows

**Additional context**

**Comment by brondani** [comment]
Currently the object file extensions are fully handled by CMake.
For IAR it's in the IAR.cmake module:
https://gitlab.kitware.com/cmake/cmake/-/blob/master/Modules/Compiler/IAR.cmake#L34
```
set(CMAKE_${lang}_OUTPUT_EXTENSION ".o")
```

@jlonnberg For better alignment between IDE IAR and CMake maybe you would like to consider updating the CMake upstream to enable the `CMAKE_${lang}_OUTPUT_EXTENSION_REPLACE` variable?

**Comment by jlonnberg** [comment]
We'll look into it!

**Comment by jkrech** [comment]
@tcsunhao can you please confirm that this is not a recent regression of cbuildgen.
@brondani and @jlonnberg is there a workaround which we can make in the IAR.<version>.cmake before a new version of CMake could fix this?

**Comment by brondani** [comment]
A possible action is to enable the `CMAKE_${lang}_OUTPUT_EXTENSION_REPLACE` variables in the toolchain file [IAR.\<version>.cmake](https://github.com/Open-CMSIS-Pack/devtools/blob/main/tools/buildmgr/cbuildgen/config/IAR.9.32.1.cmake) like this:
```
# Enable object files extension replacement
set(CMAKE_ASM_OUTPUT_EXTENSION_REPLACE 1)
set(CMAKE_C_OUTPUT_EXTENSION_REPLACE 1)
set(CMAKE_CXX_OUTPUT_EXTENSION_REPLACE 1)
```
@jlonnberg Would you agree with this until you occasionally decide to have it in the upstream?

**Comment by tcsunhao** [comment]
Hi @jkrech , sorry I cannot confirm that. We have never met such issue until 4 days ago.

**Comment by jkrech** [comment]
@tcsunhao did you change the version of CMake? CMSIS-Toolbox? Something must have changed if the output is different now.

**Comment by jkrech** [comment]
@tcsunhao  thanks for clarifying. It is clear now.

**Comment by jlonnberg** [comment]
@tcsunhao Could you test if @brondani's solution with replacing the extensions work? It should be placed just beforce the call to project(...) to work. If that works, we can add is as a WA for now and I'll rework the cmake support and we can remove such a fix when the next cmake release is out.

**Comment by tcsunhao** [comment]
@jlonnberg , I haved updated IAR.9.32.1.cmake by adding the lines from @brondani's comment
![image](https://user-images.githubusercontent.com/3381677/236388554-09bfe0d0-6fb9-45ec-8577-b2c1396f19e5.png)

It works. The .c file is compiled into .o file without '.c' inside.

**Comment by jlonnberg** [comment]
@tcsunhao Great! We'll have some discussions on this internally to see that fits out model best and the solve the problem for real

**Comment by jkrech** [comment]
Note this 'fix' will be included starting CMSIS-Toolbox 1.7.0 for the time being.
